pypeit/core/flat.py

```diff
@@ -184,6 +177,8 @@ def fit_flat(flat, tilts_dict, tslits_dict_in, slit, slitmask_func = Spectrograp
     3-Sep-2018 Ported to python by J. F. Hennawi and significantly improved
     """
 
+    spectrograph = load_spectrograph(spectrograph)
```

Comment:
If spectrograph is None, it will call load_spectrograph with spectrograph None, which I changed to return the generic spectrograph. So yes it works.